src/OrchardCore.Modules/OrchardCore.Menu/Migrations.cs

```diff
@@ -14,10 +14,16 @@ public Migrations(IRecipeMigrator recipeMigrator)
         }
 
         public async Task<int> CreateAsync()
+        {
+            // Return 1 to shortcut the second migration on new content definition schemas.
+            return await Task.FromResult(1);
+        }
+
+        public async Task<int> UpdateFrom1Async()
         {
             await _recipeMigrator.ExecuteAsync("menu.recipe.json", this);
 
```

Comment:
And here you need to create a second recipe `content-menu-picker-menu.recipe.json` which only has the definitions required for this new feature. 
Does that make sense?